lib/optimize/SplitChunksPlugin.js

```diff
@@ -162,192 +307,229 @@ const totalSize = sizes => {
+	if (typeof cacheGroups === "function") {
+		return cacheGroups;
+	}
+	if (typeof cacheGroups === "object" && cacheGroups !== null) {
+		/**
+		 * @param {Module} module the current module
+		 * @param {CacheGroupsContext} context the current context
+		 * @returns {CacheGroupSource[]} the matching cache groups
+		 */
+		const fn = (module, context) => {
+			/** @type {CacheGroupSource[]} */
+			let results = [];
+			for (const key of Object.keys(cacheGroups)) {
+				let option = cacheGroups[key];
+				if (option === false) {
+					continue;
+				}
+				if (typeof option === "string" || option instanceof RegExp) {
+					if (checkTest(option, module, context)) {
+						results.push(
+							createCacheGroupSource({
+								key
+							})
+						);
 					}
-					if (typeof option === "function") {
-						let result = option(module);
-						if (result) {
-							if (results === undefined) results = [];
-							for (const r of Array.isArray(result) ? result : [result]) {
-								const result = Object.assign({ key }, r);
-								if (result.name) result.getName = () => result.name;
-								if (result.chunks) {
-									result.chunksFilter = SplitChunksPlugin.normalizeChunksFilter(
-										result.chunks
-									);
-								}
-								result.minSize = normalizeSizes(result.minSize);
-								result.maxSize = normalizeSizes(result.maxSize);
-								results.push(result);
-							}
+				} else if (typeof option === "function") {
+					const getCacheGroups = /** @type {OptimizationSplitChunksGetCacheGroups} */ (option);
+					const result = getCacheGroups(module);
+					if (result) {
+						const groups = Array.isArray(result) ? result : [result];
+						for (const group of groups) {
+							results.push(
+								createCacheGroupSource(Object.assign({ key }, group))
+							);
 						}
-					} else if (
-						SplitChunksPlugin.checkTest(option.test, module, context) &&
-						SplitChunksPlugin.checkModuleType(option.type, module)
+					}
+				} else {
+					if (
+						checkTest(option.test, module, context) &&
+						checkModuleType(option.type, module)
 					) {
-						if (results === undefined) results = [];
-						results.push({
-							key: key,
-							priority: option.priority,
-							getName:
-								SplitChunksPlugin.normalizeName(option.name || name) ||
-								(() => {}),
-							chunksFilter: SplitChunksPlugin.normalizeChunksFilter(
-								option.chunks
-							),
-							enforce: option.enforce,
-							minSize: normalizeSizes(option.minSize),
-							maxSize: normalizeSizes(option.maxSize),
-							minChunks: option.minChunks,
-							maxAsyncRequests: option.maxAsyncRequests,
-							maxInitialRequests: option.maxInitialRequests,
-							filename: option.filename,
-							idHint: option.idHint,
-							automaticNameDelimiter: option.automaticNameDelimiter,
-							reuseExistingChunk: option.reuseExistingChunk
-						});
+						results.push(
+							createCacheGroupSource({
```

Comment:
Looks like this could be simplified to 

``` js
Object.assign({}, option, { key, name: option.name || name }
```

Reply:
Maybe even `Object.assign({ key }, option)`

The name defaulting is not needed here, as it's done below.